### PR TITLE
refactor: refactor the container manager

### PR DESCRIPTION
**Pull request by skoo87**
define a "Container" structure, it represents the container instance in
pouch engine. The "Container"s are stored in memory, so we can use
"Container".Lock() to protect itself instead of the "KMutex", because
there is defects that using "KMutex"

Signed-off-by: skoo87 <marckywu@gmail.com>

### CODE REVIEW

**Comment by allencloud** [comment]
CI fails:
```
daemon/mgr/container.go:130:1: receiver name cm should be consistent with previous receiver name mgr for ContainerManager
Makefile:44: recipe for target 'lint' failed
make: *** [lint] Error 1
```
@skoo87

**Comment by skoo87** [comment]
please review this, ping @allencloud .

**Comment by allencloud** [comment]
OK, I will review this PR this afternoon. I am a little bit busy on some other things. @skoo87

**Comment by skoo87** [comment]
The PR will affect the next step @allencloud

**Comment by allencloud** [comment]
I will finish this morning. Sorry for the delay.

File: daemon/mgr/container_types.go [hunk 1]
```diff
@@ -17,3 +20,29 @@ type ContainerRemoveOption struct {
 	Volume bool
 	Link   bool
 }
+
+// Container represents the container instance in runtime.
+type Container struct {
+	sync.Mutex
+	meta *types.ContainerInfo
```
Comment (allencloud) [inline daemon/mgr/container_types.go]: Why not just use `types.ContainerInfo` directly?
Is it related to the pointer which could save some memory?

**Comment by allencloud** [comment]
Look Good to Me, unless my confusion in the commit comment is answered. @skoo87

**Comment by allencloud** [comment]
LGTM

**Pull request merged**